module/shop_beta/shop_guild.py

```diff
@@ -0,0 +1,291 @@
+                List types exclusively for PR/DR series selection
+
+        Returns:
+            Button:
+
+        Raises:
+            ScriptEnd:
+        """
+        # Ensure there is valid SELECT_* combination for item group
+        # Stow variables for frequent usage
+        lgroup = item.group.lower()
+        ugroup = item.group.upper()
+        try:
+            choices = globals()[f'SELECT_{ugroup}']
+        except KeyError:
+            logger.critical(f'Missing SELECT_{ugroup} (global)')
+            raise ScriptEnd
+
+        # Choose the applicable SELECT_GRID_* for item group
+        shop_select_grid = None
+        if lgroup == 'book':
+            shop_select_grid = SELECT_GRID_3X1
+        elif lgroup == 'box' or lgroup == 'retrofit':
+            shop_select_grid = SELECT_GRID_4X1
+        elif lgroup == 'pr' and isinstance(choice, list):
+            # Complex grid retrieval based on PR/DR series
+            # Determine series through asset appearance
+            # and convert choice from list(str) to str
+            for idx, button in enumerate(SHOP_SELECT_PR):
+                if self.appear(button, offset=(20, 20)):
+                    choice = choice[idx]
+                    if idx == 0:
+                        shop_select_grid = SELECT_GRID_6X1
+                    else:
+                        shop_select_grid = SELECT_GRID_4X1
+                    break
+        elif lgroup == 'plate':
+            shop_select_grid = SELECT_GRID_5X1
+
+        if shop_select_grid is None:
+            logger.warning(f'Failed to find applicable grid for group \'{lgroup}\'')
+            return None
+
+        # Return button from appropriate grid based on 'choice'
+        if choice in choices:
+            return shop_select_grid.buttons[choices.get(choice)]
+
+        logger.critical(f'Missing \'{choice}\' in SELECT_{ugroup}')
+        raise ScriptEnd
+
+    def shop_buy_select_execute(self, item):
+        """
+        Args:
+            item: Item to check
+
+        Returns:
+            bool: implicating failed to execute
+
+        Raises:
+            ScriptEnd:
+        """
+        # Base case - Only items with 'postfix' are allowed
+        if not hasattr(item, 'postfix'):
+            logger.critical('Unexpected item object; missing \'postfix\' attr')
+            raise ScriptEnd
+
+        # Retrieve appropriate globals and config values for processing
+        # Stow variables for frequent usage
+        ugroup = item.group.upper()
+        postfix = item.postfix
+        try:
+            limit = globals()[f'SELECT_{ugroup}_LIMIT']
+            choice = getattr(self.config, f'GuildShop_{ugroup}{postfix}')
+        except:
+            logger.critical('Missing either or both of the following:')
+            logger.critical(f'- SELECT_{ugroup}_LIMIT (global)')
+            logger.critical(f'- GuildShop_{ugroup}{postfix} (config)')
+            raise ScriptEnd
```

Comment:
Abstract a function that receives an `item` and returns `limit` and `choice`, remove attribute `postfix`, move the creation of `postfix` in `GuildItemGrid.predict()` into this function. Raise `ScriptError` instead of `ScriptEnd`.

Reply:
Went a different direction with this but I incorporated a bit of the comment as well.

Compress all those globals I had previously into a dictionary and squash that large if/else clause I had. Seems straight forward, will just have to try it to make sure it goes fine.

Reply:
Looking good.